proposals/0002-root-signature-in-clang.md

```diff
@@ -936,6 +936,159 @@ Operands:
 * Does this require additional hardware/software/human resources?
 -->
 
+### Root Signature Driver Options
+
+A user can compile and use a root signature for a shader using the different
+command line options. Below lists the options available in DXC that will be
+carried forward to Clang and their expected behaviour.
+
+(Design notes and unplanned options will be removed after being resolved)
+
+#### Option `-force-rootsig-ver`
+
+Compiles the root signatures with a specific root signature version.
+
+Usage:
+
+```
+  -force-rootsig-ver rootsig_1_0
+  -force-rootsig-ver rootsig_1_1
+```
+
+Behaviour:
+
+ - Validation logic that is specific to the version will be followed, as
+described above
+ - Serialized root signature in DXIL container will be compliant with specified
+version
+
+Design Notes: Required for backwards compatibility. Implemented in Clang.
+
+#### Option `-rootsig-define`:
+
+Sets the root signature attribute for the entry function to be the root
+signature defined by the given macro expansion.
+
+Usage:
+
+Given a defined macro either provided in the source file
+(`#define RS "CBV(b0)"`), or, as a command line define
+(`-D RS="CBV(b0)"`)
+
+```
+  -rootsig-define RS
+```
+
+Behaviour:
+
+ - If the entry function does not have a root signature attribute, it will use
+the one defined by the macro expansion
+
+ - If the entry function has a root signature attribute, it will overwrite to
+use the one defined by the macro expansion
+
+Design Notes: Commonly used. Planned in Clang.
+
+#### Target Root Signature Version
+
+Compiles the "entry" root signature, specified by the given macro expansion, to
+a DXIL Container with just the (version specific) RTS0 part.
+
+Usage:
+
+Given a defined macro in the source file (`#define RS "CBV(b0)"`)
```

Comment:
Perhaps we should note that it's not necessary to specify the root signature macro in the source file, since it may be specified with a `-D` option on the command line instead.  DXC supports this, but doesn't accept an empty source file. The source file will not even be used in this case, so it could in theory be omitted, but DXC requires it, and it seems to require that it has at least 4 bytes of content (that's likely due to the BOM detection).